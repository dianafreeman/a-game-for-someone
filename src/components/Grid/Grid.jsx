import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Plane from '../Plane';

const Row = ({ y, length, onClick }) => {
  const r = [];
  for (let x = 0; x < length; x += 1) {
    r.push(<Plane onClick={onClick} xPos={x} yPos={y} key={x} />);
  }
  return r;
};

/*
 size = 3
 origin = 0,0

 sideLength = (size - 1) / 2
 OriginRow = -sideLength -> [0,0] => +sideLength
 */
const zeroToHalf = (half) => {
  const out = [];
  for (let i = 0; i < (half / 2); i += 1) {
    out.push(i);
  }
  return out;
};

const negToPos = (pos) => {
  const reversed = [...pos].reverse();
  const neg = reversed.slice(0, -1).map((i) => -i);
  return [...neg, ...pos];
};


// const cell = (x, y) => [x, y];
const placements = [
  [
    [-1, -1], [0, -1], [1, -1],
  ],
  [
    [-1, 0], [0, 0], [1, 0],
  ],
  [
    [-1, 1], [0, 1], [1, 1],
  ],
];

const makeXs = (size) => negToPos(zeroToHalf((size - 1 / 2)));

const RowData = (xSize, yPosition) => {
  const xS = makeXs(xSize);
  return xS.map((x) => [x, yPosition]);
};

function Grid({ size = 5, onSquareClick }) {
  return (
    <>
      {placements.map((row) => row.map((cell) => <Plane onClick={onSquareClick} xPos={cell[0]} yPos={cell[1]} key={`plane-row-${row}-cell-${cell}`} />))}
    </>
  );
}

Grid.propTypes = {
  size: PropTypes.number.isRequired,
  onSquareClick: PropTypes.func.isRequired,
};

export default Grid;
