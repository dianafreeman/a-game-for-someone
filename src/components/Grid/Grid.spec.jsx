import React from 'react';
import { render } from '@testing-library/react';
import Grid from '.';

describe('Grid', () => {
  it('renders without crashing, matches snapshot', () => {
    const { container } = render(<Grid size={6} />);
    expect(container).toMatchSnapshot();
  });
  describe.only('size from grid', () => {
    it('renders without crashing, matches snapshot', () => {
      const size = 7;
      const thing = render(<Grid size={size} />);

      expect(thing).toEqual(6);
    });
  });
});
