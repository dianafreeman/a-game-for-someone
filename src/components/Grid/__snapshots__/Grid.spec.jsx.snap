// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders without crashing, matches snapshot 1`] = `
<div>
  <mesh
    position="0,0,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="1,0,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="2,0,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="3,0,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="4,0,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="5,0,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="0,1,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="1,1,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="2,1,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="3,1,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="4,1,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="5,1,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="0,2,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="1,2,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="2,2,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="3,2,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="4,2,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="5,2,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="0,3,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="1,3,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="2,3,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="3,3,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="4,3,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="5,3,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="0,4,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="1,4,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="2,4,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="3,4,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="4,4,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="5,4,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="0,5,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="1,5,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="2,5,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="3,5,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="4,5,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
  <mesh
    position="5,5,0"
  >
    <planebuffergeometry
      args="1,1,1"
      attach="geometry"
    />
    <meshphysicalmaterial
      attach="material"
      color="gray"
    />
  </mesh>
</div>
`;
